sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -112,20 +112,23 @@ case class AnalyzeCreateTable(sparkSession: SparkSession) extends Rule[LogicalPl
         throw new AnalysisException("Saving data into a view is not allowed.")
       }
 
-      if (DDLUtils.isHiveTable(existingTable)) {
-        throw new AnalysisException(s"Saving data in the Hive serde table $tableName is " +
-          "not supported yet. Please use the insertInto() API as an alternative.")
+      val (isProviderMatch, existingProvider, specifiedProvider) =
+        DDLUtils.isHiveTable(existingTable) match {
+        case false =>
+          val existing = DataSource.lookupDataSource(existingTable.provider.get)
+          val specified = DataSource.lookupDataSource(tableDesc.provider.get)
+          (existing == specified, existing.getSimpleName, specified.getSimpleName)
+        case true =>
+          val existing = existingTable.provider.get
+          val specified = tableDesc.provider.get
+          (existing == specified, existing, specified)
       }
-
       // Check if the specified data source match the data source of the existing table.
-      val existingProvider = DataSource.lookupDataSource(existingTable.provider.get)
```

Comment:
We have `HiveFileFormat`, and we can make it implement `DataSourceRegister`, then `DataSource.lookupDataSource("hive")` can work.